test/targets.cpp

```diff
@@ -50,4 +51,21 @@ TEST_CASE(targets)
     EXPECT(ts.size() >= 1);
 }
 
+TEST_CASE(concurrent_targets)
+{
+    std::vector<std::thread> threads;
+
+    for(auto i = 0u; i < 1000; i++)
+    {
+        auto thread_body = []() {
+            auto ref_target = migraphx::make_target("ref");
+            migraphx::register_target(ref_target);
+        };
+
+        threads.emplace_back(thread_body);
+    }
+    for(auto& tt : threads)
+        tt.join();
```

Comment:
Use `joinable_thread` instead of manually joining threads.

Reply:
Can you clarify?  That's not a class or function name.

Reply:
Its a class here: https://github.com/ROCmSoftwarePlatform/AMDMIGraphX/blob/develop/src/include/migraphx/par_for.hpp#L36

Reply:
done